### ISSUE TITLE
Fail to import asset by url when current script src has query params

### ISSUE THREAD

**Issue by edoardocavazza** [rollup/rollup#2603]
- Rollup Version: 0.66.6
- Operating System (or Browser): MacOS 10.14.1
- Node Version: 11.4.0

### How Do We Reproduce?

```js
// index.js
import IMAGE_URL from './image.png';

console.log(IMAGE_URL);
```
```js
// rollup.config.js
export default {
    input: './index.js',
    output: {
        file: './bundle.js',
        format: 'umd',
    },
    plugins: [{
        load(id) {
            if (!id.endsWith('.png')) {
                return null;
            }
            let buffer = require('fs').readFileSync(id);
            let asset = this.emitAsset(path.basename(id), buffer);
            return `export default import.meta.ROLLUP_ASSET_URL_${asset};`;
        }
    }],
}
```

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Document</title>
</head>
<body>
    <script src="bundle.js?test=1"></script>
</body>
</html>
```

### Expected Behavior
The `IMAGE_URL` value is `http://localhost/image.png`

### Actual Behavior
The `IMAGE_URL` value is `http://localhost/bundle.js?test=1/../image.png`

**Comment by lukastaegert** [comment]
I think this should be fixable by using the `URL` constructor a little more intelligently but I need to check.

**Comment by lukastaegert** [comment]
Fix at #2937
